kalatori-ah/src/lib.rs

```diff
@@ -0,0 +1,322 @@
+        .as_str()
+    {
+        "USDC" => Usd::C,
+        "USDT" => Usd::T,
+        _ => anyhow::bail!("{USD_ASSET} must equal USDC or USDT"),
+    };
+
+    let pair = Pair::from_string(
+        &env::var(SEED).with_context(|| format!("`{SEED}` isn't set"))?,
+        None,
+    )
+    .with_context(|| format!("failed to generate a key pair from `{SEED}`"))?;
+
+    let endpoint = env::var(RPC).or_else(|error| {
+        if error == VarError::NotPresent {
+            log::debug!(
+                "`{RPC}` isn't present, using the default value instead: \"{DEFAULT_RPC}\"."
+            );
+
+            Ok(DEFAULT_RPC.into())
+        } else {
+            Err(error).context(format!("failed to read `{RPC}`"))
+        }
+    })?;
+
+    let override_rpc = env::var_os(OVERRIDE_RPC).is_some();
+
+    let database_path = if env::var_os(IN_MEMORY_DB).is_none() {
+        Some(env::var(DATABASE).or_else(|error| {
+            if error == VarError::NotPresent {
+                let default_v = match usd_asset {
+                    Usd::C => "database-ah-usdc.redb",
+                    Usd::T => "database-ah-usdt.redb",
+                };
+
+                log::debug!(
+                    "`{DATABASE}` isn't present, using the default value instead: \"{default_v}\"."
+                );
+
+                Ok(default_v.into())
+            } else {
+                Err(error).context(format!("failed to read `{DATABASE}`"))
+            }
+        })?)
+    } else {
+        if env::var_os(DATABASE).is_some() {
+            log::warn!(
+                "`{IN_MEMORY_DB}` is set along with `{DATABASE}`. The latter will be ignored."
+            );
+        }
+
+        None
+    };
+
+    let destination = match env::var(DESTINATION) {
+        Ok(destination) => Ok(Some(
+            AccountId32::try_from(hex::decode(&destination[2..])?.as_ref())
+                .map_err(|()| anyhow::anyhow!("unknown destination address length"))?,
+        )),
+        Err(VarError::NotPresent) => Ok(None),
+        Err(error) => Err(error).context(format!("failed to read `{DESTINATION}`")),
+    }?;
+
+    log::info!(
+        "Kalatori {} by {} is starting...",
+        env!("CARGO_PKG_VERSION"),
+        env!("CARGO_PKG_AUTHORS")
+    );
+
+    let shutdown_notification = CancellationToken::new();
+    let (error_tx, mut error_rx) = mpsc::unbounded_channel();
+
+    let (api_config, endpoint_properties, updater) =
+        rpc::prepare(endpoint, shutdown_notification.clone(), usd_asset)
+            .await
+            .context("failed to prepare the node module")?;
+
+    let (database, last_saved_block) = Database::initialise(
+        database_path,
+        override_rpc,
+        pair,
+        endpoint_properties,
+        destination,
+    )
+    .context("failed to initialise the database module")?;
+
+    let processor = Processor::new(api_config, database.clone(), shutdown_notification.clone())
```

Comment:
>`database(clone)`

Right now it seems that this works OK concurrently, but later it would be best to move db handler into separate thread and talk to it through messages to isolate it (and make it even faster by disk write yelds).

I would think that in times of refactor this is one of the highest priority things.